src/Controller/TranslationsController.php

```diff
@@ -0,0 +1,172 @@
+        }
+
+        $this->set('schema', $this->Schema->getSchema($this->objectType));
+
+        $object = Hash::extract($response, 'data');
+        $this->set('translation', $translation);
+        $this->set('object', $object);
+
+        return null;
+    }
+
+    /**
+     * Create or edit single translation.
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function save() : ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->prepareRequest($this->objectType);
+        $objectId = $requestData['object_id'];
+        if (!empty($requestData['id'])) {
+            unset($requestData['object_id']);
+        }
+        $lang = $requestData['lang'];
+        try {
+            $this->apiClient->save('translations', $requestData);
+        } catch (BEditaClientException $e) {
+            // Error! Back to object view or index.
+            $this->log($e, LogLevel::ERROR);
+            $this->Flash->error($e, ['params' => $e->getAttributes()]);
+
+            if ($this->request->getData('id')) {
+                return $this->redirect([
+                    '_name' => 'translations:edit',
+                    'object_type' => $this->objectType,
+                    'id' => $objectId,
+                    'lang' => $lang,
+                ]);
+            }
+
+            return $this->redirect([
+                '_name' => 'translations:add',
+                'object_type' => $this->objectType,
+                'id' => $objectId,
+            ]);
+        }
+
+        return $this->redirect([
+            '_name' => 'translations:edit',
+            'object_type' => $this->objectType,
+            'id' => $objectId,
+            'lang' => $lang,
+        ]);
+    }
+
+    /**
+     * Delete translation(s).
+     *
+     * @return \Cake\Http\Response|null
+     */
+    public function delete() : ?Response
+    {
+        $this->request->allowMethod(['post']);
+        $requestData = $this->request->getData();
+        foreach ($requestData as $translation) {
+            if (!empty($translation['id'])) {
+                try {
+                    // remove completely the translation
+                    $this->apiClient->delete(sprintf('/translations/%s', $translation['id']));
+                } catch (BEditaClientException $e) {
+                    $this->log($e, LogLevel::ERROR);
+                    $this->Flash->error($e, ['params' => $e->getAttributes()]);
+
+                    // redir to main object view
+                    return $this->redirect(['_name' => 'modules:view', 'object_type' => $this->objectType, 'id' => $translation['object_id']]);
+                }
+            }
+        }
+        $this->Flash->success(__('Translation(s) deleted'));
+
+        // redir to referer
+        return $this->redirect(['_name' => 'modules:view', 'object_type' => $this->objectType, 'id' => $translation['object_id']]);
```

Comment:
[see scrutinizer note above]

I'd use this rule in case of single delete (`$requestData` count is 1) - otherwise I'll redirect to `$this->object_type` list